packages/ui/src/index.ts

```diff
@@ -13,7 +13,7 @@ export { Flex, type FlexProps } from './components/Flex';
 export { Grid, type GridProps } from './components/Grid';
 export { Heading, type HeadingProps } from './components/Heading';
 export { Image, type ImageProps } from './components/Image';
-export { Input, type InputProps } from './components/Input';
+export { Input, type InputProps, type IconType } from './components/Input';
```

Comment:
Rename `IconType` to a name more explicitly, like `IconTypeProp.` Also, maybe it's better to create a hook name  `useIconElement` that contains the `renderIcon` and the `useMemo.`